src/main/scala/za/co/absa/abris/avro/sql/AvroDataToCatalyst.scala

```diff
@@ -116,10 +128,14 @@ case class AvroDataToCatalyst(
     val length = buffer.limit() - 1 - ConfluentConstants.SCHEMA_ID_SIZE_BYTES
     decoder = DecoderFactory.get().binaryDecoder(buffer.array(), start, length, decoder)
 
-    val writerSchema = getWriterSchema(schemaId)
-    reader = new GenericDatumReader[Any](writerSchema, avroSchema)
+    val reader = Option(confluentReaderCache.get(schemaId)).getOrElse{
```

Comment:
`confluentReaderCache.computeIfAbsent(...)` could be more readable and concise here

Reply:
I used a similar method with Scala's HashMap now.